### DISCUSSION TITLE
Please start pushing bug fixes to release and keep new features in dev

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [OpenRCT2/OpenRCT2#16781] [Feature requests]
Please change your update policy.
It would be nice if bug fixes were pushed to stable as patch versions and features were kept minor or major versions.

Adding new features adds more bugs, so its not good to keep them in the same update.

**Comment by 733737** [comment]
the nsf branch is taking all of the time of the devs, so I don't know if they can really run two different branches like you say

**Reply by Shadowblitz16** [reply]
The point is so we can have a very stable build, it's the same reason there are LTS versions of operating systems and software.

I'm not asking for a LTS release though, I am just asking that the current stable receives bugfixes from dev until the dev build becomes the new stable, at which point the old stable stop receiving bug patches

**Reply by ocalhoun6** [reply]
But it's still asking the developers to maintain _two_ branches instead of one.

Currently, the release branch is just older versions of the develop branch, so the only branch actually being maintained is the develop branch.

But when you start talking about updating the release branch with bug fixes but _not_ updating it with new features ... then it becomes a distinct branch of its own, which is not identical to any develop branch that ever existed.

So then you have two branches being maintained ... along with all the headaches that sort of thing can bring.


-----

Honestly, the best way to go is to simply run the develop branch all the time. _Occasionally_ a new feature will introduce a new bug ... but when that happens, it's usually solved very quickly in the next nightly update or two.

**Comment by frutiemax** [comment]
The problem with this is that the QA is mostly done by regular users, and there will be unspotted bugs that get through the master release. Also keep in mind that any code that is modified can introduce new bugs, not just features.

**Comment by Broxzier** [comment]
Pushing fixes to the release branch adds a lot of overhead. It is not a matter of simply applying the fix from the develop branch to the release branch, since code that needs fixing can differ between branches or be added in new places that don't even exist in the other branch. This only increases the risk of partly fixing issues in either branch. Not to mention how it's only even more work (even impossible) to apply a fix for something that has been refactored.
